term_proposal/YSHsieh7777/README.md

```diff
@@ -28,6 +28,56 @@ People often fall into **panic** when they meet dangers like fire or earthquake.
  - Calculate the number of people that die in the fire. 
  - Store the **best** traffic flow.
 
+## Project Structure
+
+Written with `C++` because it needs a large amount of computations. If it's hard to visualize the simulation with `C++`, `C++`(for large amount of computations) + `Pybind11` + `Python`(for user input processing and visualization).
+
+The program will need a **long loop** to update map and people's conditions, so a very efficient way to update conditions is necessary. It means that I need a well data structure that is easy to traverse, compute and update. Also, parallel programming is a possible strategy to improve performance.
+
+### Libraries
+ - Using [SDL](https://www.libsdl.org/) in `C++` or [PYGame](https://www.pygame.org/news) in `Python` for visualization.
+ - Using `SDL` in `C++` or `PYGame` in `Python` to get user inputs for constructing map.
+ - Using [class](http://www.cplusplus.com/doc/tutorial/classes/), [vector](http://www.cplusplus.com/reference/vector/vector/), [map](http://www.cplusplus.com/reference/map/map/?kw=map) and [pair](http://www.cplusplus.com/reference/utility/pair/?kw=pair) to build a nice data structure.
+ - Using [OpenMP](https://www.openmp.org/) for parallel programming or [OpenMP with Cython](https://gist.github.com/zed/2051661). I think [CUDA](https://devblogs.nvidia.com/even-easier-introduction-cuda/) is hard to implement because there are many `conditional operator` when updating people's conditions.
+
+### Data structure
+
+The following are some ideas about data structures that may be used. These only meet requirements but not optimal.
+
+```c++
+class Indoor_map
```

Comment:
Are you assuming a Cartesian grid?

Anyway, the grid design may be the most fundamental part in the system.  It determines the system capability.  I don't think you can reach a perfect design within just 2 months, but it's worth thinking.